src/highdicom/seg/sop.py

```diff
@@ -568,17 +570,6 @@ def __init__(
         self.NumberOfFrames = 0
         self.PerFrameFunctionalGroupsSequence: List[Dataset] = []
 
-        if pixel_array.ndim == 2:
-            pixel_array = pixel_array[np.newaxis, ...]
-        if pixel_array.ndim not in [3, 4]:
-            raise ValueError('Pixel array must be a 2D, 3D, or 4D array.')
-
-        if pixel_array.shape[1:3] != (self.Rows, self.Columns):
-            raise ValueError(
-                'Pixel array representing segments has the wrong number of '
-                'rows and columns.'
-            )
```

Comment:
This is true by construction of the self.Rows and self.Columns, so the check is pointless